### PR TITLE
[CAS-327] MessagesHeaderView new design

### PR DESCRIPTION
**Pull request by samiuelson**
https://stream-io.atlassian.net/browse/CAS-327

### Description

Add `MessagesListHeader` view to `stream-chat-android-ui-components` module.

Screenshot 2020-11-19 at 11 57 43

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/header/MessagesHeaderView.kt [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+import androidx.core.content.res.use
+import androidx.core.view.isVisible
+import com.getstream.sdk.chat.style.TextStyle
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamMessagesHeaderViewBinding
+import io.getstream.chat.android.ui.utils.extensions.getDimension
+
+public class MessagesHeaderView : ConstraintLayout {
```
Comment (samiuelson) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/header/MessagesHeaderView.kt:23]: We've voted to name this class in the `Channel...` manner. However, there is already `ChannelsView` so `MessagesHeaderView` sounds less confusing for me.

Comment (zsmb13) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/header/MessagesHeaderView.kt:23]: Oh, these are gonna be messy 😄 

I think our best chance of fixing up the naming at this point is taking a look at all the classes we have when we're done with the new design, and then renaming / resorting them into packages.

**Review by zsmb13 — APPROVED** [review]
LGTM

**Review by zsmb13 — APPROVED** [review]
Still looks good to me! 😄

**Pull request merged**